doc/docs/modules/ROOT/pages/quickstart_writer.adoc

```diff
@@ -215,4 +215,88 @@ UNwIND $events AS event
 CREATE (n:Person{fullName: event.name + event.surname})
 ----
 
-Where `event` represents each Dataset row.
\ No newline at end of file
+Where `event` represents each Dataset row.
+
+==== Schema Optimization Operations
+
+The spark connector supports schema optimization operations via:
+
+* index
+* constraints
+* set of schema queries
+
+that will be executed *before* the import process will start in order to speed-up the import itself.
+
+You can set the optimization via `schema.optimization.type` option that takes three values:
+
+* `INDEX`: it creates only indexes on provided nodes
+* `CONSTRAINTS`: it creates only indexes on provided nodes
+* `QUERY`: it perform a series of schema queries separated by `;`
+
+and it works only when you're merging nodes.
+
+===== Index Creation
+
+Following an example of how to create indexes while you're creating nodes
+
+----
+ds.write
+      .format(classOf[DataSource].getName)
+      .mode(SaveMode.Overwrite)
+      .option("url", SparkConnectorScalaSuiteIT.server.getBoltUrl)
+      .option("labels", ":Person:Customer")
+      .option("node.keys", "surname")
+      .option("schema.optimization.type", "INDEX")
+      .save()
+----
+
+This will create, before the import starts, the following schema query:
+
+----
+CREATE INDEX ON :Person(surname)
+----
+
+*So please into consideration that the first label is used for the index creation*
+
+
+===== Constraint Creation
+
+Following an example of how to create indexes while you're creating nodes
+
+----
+ds.write
+      .format(classOf[DataSource].getName)
+      .mode(SaveMode.Overwrite)
+      .option("url", SparkConnectorScalaSuiteIT.server.getBoltUrl)
+      .option("labels", ":Person:Customer")
+      .option("node.keys", "surname")
+      .option("schema.optimization.type", "CONSTRAINT")
+      .save()
+----
+
+This will create, before the import starts, the following schema query:
```

Comment:
I am not sure about this approach for a couple of reasons:

- If the number of nodes is large, creating the index before you begin a write guarantees that index population starts, not that the index is ready to go, unless you poll/monitor the index
- User doesn't practically have a way of controlling index naming
- Complex logic....if an index already exists from outside of spark, this should make 2 of them with double the overhead for the database.
- If spark creates indexes for your automatically, there are already a number of kinds of index syntax not yet supported, and in the future there will be more.  This is an additional point of coupling to the database language, and it can change.    For example, what should happen if someone does a fancy new 4.2 supported index type on a 3.5 database when the spark surface makes it look you can do this?

Reply:
> I am not sure about this approach for a couple of reasons:
> 
> * If the number of nodes is large, creating the index before you begin a write guarantees that index population starts, not that the index is ready to go, unless you poll/monitor the index

We can add a future option that waits for index to or we can add it in this PR if you want

> * User doesn't practically have a way of controlling index naming

True, but we use a naming convention `spark_<index_or_constraint>_<label>` making it recognisable
 
> * Complex logic....if an index already exists from outside of spark, this should make 2 of them with double the overhead for the database.

Sorry I don't get this can you elaborate?

> * If spark creates indexes for your automatically, there are already a number of kinds of index syntax not yet supported, and in the future there will be more.  This is an additional point of coupling to the database language, and it can change.    For example, what should happen if someone does a fancy new 4.2 supported index type on a 3.5 database when the spark surface makes it look you can do this?

This is just syntax sugar which is meant to simplify the index/constraint creation fro  for the user but we also are future proof because there is always the `QUERY` option where the developer can specify a set of schema write queries like these:

```scala
ds.write
      .format(classOf[DataSource].getName)
      .mode(SaveMode.Overwrite)
      .option("url", SparkConnectorScalaSuiteIT.server.getBoltUrl)
      .option("labels", ":Person:Customer")
      .option("node.keys", "surname")
      .option("schema.optimization.type", "QUERY")
      .option("schema.optimization.query", """CREATE INDEX ON :Person(surname);
          |CREATE CONSTRAINT ON (p:Product)
          | ASSERT (p.name, p.sku)
          | IS NODE KEY
          |""".stripMargin)
      .save()
```